internal/cloud/azure/azure.go

```diff
@@ -388,6 +446,89 @@ func (c *Cloud) getLoadBalancerDNSName(ctx context.Context) (string, error) {
 }
 */
 
+// PrepareAzureControlPlaneNode sets up iptables for the control plane node only
+// if an internal load balancer is used.
+//
+// This is needed since during `kubeadm init` the API server must talk to the
+// kubeAPIEndpoint, which is the load balancer IP address. During that time, the
+// only healthy VM is the VM itself. Therefore, traffic is sent to the load balancer
+// and the 5-tuple is (VM IP, <some port>, LB IP, 6443, TCP).
+// Now the load balancer does not re-write the source IP address only the destination (DNAT).
+// Therefore the 5-tuple is (VM IP, <some port>, VM IP, 6443, TCP).
+// Now the VM responds to the SYN packet with a SYN-ACK packet, but the outgoing
+// connection waits on a response from the load balancer and not the VM therefore
+// dropping the packet.
+//
+// OpenShift also uses the same mechanism to redirect traffic to the API server:
+// https://github.com/openshift/machine-config-operator/blob/e453bd20bac0e48afa74e9a27665abaf454d93cd/templates/master/00-master/azure/files/opt-libexec-openshift-azure-routes-sh.yaml
+func (c *Cloud) PrepareAzureControlPlaneNode(ctx context.Context, log *logger.Logger) {
+	chainName := "azure-lb-nat"
+
+	selfMetadata, err := c.Self(ctx)
+	if err != nil {
+		log.Fatalf("failed to get self metadata: %v", err)
+	}
```

Comment:
```suggestion
	if err != nil {
		log.With(zap.Error(err).Fatalf("failed to get self metadata")
	}
```
Also applies to most other log statements in this function